src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+  static alive = [];
+
+  static animalHealth(animal) {
+    if (animal.health <= 0) {
+      Animal.alive = Animal.alive.filter((a) => a !== animal);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(herb) {
+    if (herb instanceof Herbivore && !herb.hidden) {
+      herb.health -= 50;
+
+      Animal.animalHealth(herb);
+    }
+
+    if (herb instanceof Carnivore) {
+      return herb.health;
+    }
```

Comment:
The condition checking if `herb` is an instance of `Carnivore` seems unnecessary. The `bite` method should focus on interactions between `Carnivore` and `Herbivore` according to the task description. Consider removing this condition unless there's a specific requirement for it.